src/hooks/useFetch.js

```diff
@@ -12,19 +12,34 @@ import { SETTINGS_KEYS } from '../settings';
 import { UIDatabase } from '../database';
 
 const { SYNC_SITE_NAME, SYNC_SITE_PASSWORD_HASH } = SETTINGS_KEYS;
+
 const CONNECTION_TIMEOUT_PERIOD = 10000;
 
 const ERROR_CODES = {
   ...AUTH_ERROR_CODES,
   EMPTY_RESPONSE: 'No records found',
 };
 
+const bugsnagClient = new BugsnagClient();
+
+class BugsnagError extends Error {
+  constructor(message, data, ...args) {
+    super(message, ...args);
+    bugsnagClient.notify(this, report => {
+      report.errorMessage = message;
+      report.metadata = data;
```

Comment:
not useful to retain the context of `FETCH ERROR`? 
I haven't used the bugsnag client, so don't know how useful the grouping is.

Reply:
Bugsnag shows you a snippet of the area of code which threw the exception, so I didn't really feel a need to keep it :man_shrugging:.